### PR TITLE
Add io.github.labbots.NiimPrintX

### PR DESCRIPTION
**Pull request by hadess**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.

NiimPrintX is a label design application for Niimbot Bluetooth printers.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:** https://github.com/labbots/NiimPrintX/issues/15

Still missing an appdata file before it's ready to review.

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.NiimPrintX.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+app-id: io.github.NiimPrintX
```
Comment (hfiguiere) [inline io.github.NiimPrintX.yml:1]: This appid should be `io.github.labbots.NiimPrintX`

**Comment by hadess** [comment]
I'll probably replace the screenshot once the application's merged.

**Comment by hadess** [comment]
bot, build io.github.labbots.NiimPrintX

**Comment by hadess** [comment]
bot, build io.github.labbots.NiimPrintX

**Comment by hadess** [comment]
bot, build io.github.labbots.NiimPrintX

File: poetry.yaml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        type: file
+        url: https://files.pythonhosted.org/packages/08/aa/cc0199a5f0ad350994d660967a8efb233fe0416e4639146c089643407ce6/packaging-24.1-py3-none-any.whl
+        sha256: 5b8f2217dbdbd2f7f384c41c628544e6d52f2d0f53c6d0c3ea61aa5d1d7ff124
+      - type: file
+        url: https://files.pythonhosted.org/packages/cc/20/ff623b09d963f88bfde16306a54e12ee5ea43e9b597108672ff3a408aad6/pathspec-0.12.1-py3-none-any.whl
+        sha256: a0d503e138a4c123b27490a4f7beda6a01c6f288df0e4a8b79c7eb0dc7b4cc08
+      - type: file
+        url: https://files.pythonhosted.org/packages/88/fe/90476c4f6a1b2f922efa00d26e876dd40c7279e28ec18f08f0851ad21ba6/scikit_build_core-0.10.7-py3-none-any.whl
+        sha256: 5e13ab7ca7c3c6dd019607c3a6f53cba67dade8757c4c4f75b459e2f90e4dbc3
+  - name: python3-poetry
```
Comment (bbhtt) [inline poetry.yaml:21]: is this required at runtime? if not it should have a `cleanup` with `- "*"`

File: io.github.labbots.NiimPrintX.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-stable
+command: NiimPrintX.ui
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --share=network
+  - --system-talk-name=org.bluez
+  - --env=PYTHONPATH=/app/lib/python3.12/site-packages
```
Comment (bbhtt) [inline io.github.labbots.NiimPrintX.yml:13]: is setting pythonpath necessary ?

File: poetry.yaml (hunk 2 above)
Comment (bbhtt) [reply poetry.yaml:21]: same of setupttools rust, pyinstaller etc.

**Comment by bbhtt** [comment]
can metainfo, desktop file and screenshots be moved upstream ?

**Comment by hadess** [comment]
> can metainfo, desktop file and screenshots be moved upstream ?

Trying that here:
https://github.com/labbots/NiimPrintX/pull/16

**Comment by bbhtt** [comment]
bot, build io.github.labbots.NiimPrintX

**Comment by hadess** [comment]
bot, build io.github.labbots.NiimPrintX

**Comment by hadess** [comment]
I've added more file access permissions (TkInter isn't exactly portal-ready), and started preparing for opening the filetypes it can save, but that won't work just yet: https://github.com/labbots/NiimPrintX/issues/17

File: io.github.labbots.NiimPrintX.yml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+command: niimprintx
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --share=network
+  - --system-talk-name=org.bluez
+  - --filesystem=home
+  - --filesystem=xdg-run/gvfs
+  - --filesystem=/run/media
+  - --filesystem=/media
```
Comment (bbhtt) [inline io.github.labbots.NiimPrintX.yml:16]: home + media is almost equivalent to host, probably better to use that instead

**Comment by hadess** [comment]
All the review comments were addressed. Let me know if there's anything else.

**Comment by bbhtt** [comment]
bot, build io.github.labbots.NiimPrintX

**Pull request closed without merging**